src/Dialog/Dialog.tsx

```diff
@@ -338,29 +345,13 @@ const buttonTypes = {
   danger: ButtonDanger
 }
 const Buttons: React.FC<{buttons: DialogButtonProps[]}> = ({buttons}) => {
-  const autoFocusRef = useRef<HTMLButtonElement>(null)
-  let autoFocusCount = 0
-  const [hasRendered, setHasRendered] = useState(0)
-  useEffect(() => {
-    // hack to work around dialogs originating from other focus traps.
-    if (hasRendered === 1) {
-      autoFocusRef.current?.focus()
-    } else {
-      setHasRendered(hasRendered + 1)
-    }
-  }, [hasRendered])
```

Comment:
I'm a little hesitant to remove this code because I don't fully understand why it was added. Who would have the most context about why this was necessary?

Reply:
It was added in https://github.com/primer/react/pull/1235, which @dgreif reviewed. I just noticed that PR includes [a Storybook story which demos the issue](https://github.com/primer/react/pull/1235/files#diff-1a7fe7df557c29e6117f7bbeade270a7ea033739ebe88ea1519399d94722eac7); I’m checking now whether this PR causing that story to regress.

Reply:
This PR _does_ cause the `ShorthandHookFromActionMenu` story to regress—in #1471, the `OK` button is focused, but in #1472, the `Cancel` button is focused (even though the `OK` button has `autoFocus`). I’ll close this issue in favor of #1471. 👍